### ISSUE TITLE
Keybinds on android tv box

### ISSUE THREAD

**Issue by jbarry786** [mpv-android/mpv-android#204]
Hi.

I have a android tv box, running Android 5.1.1. I have a mouse and a small remote controller.

I've tried a myriad of different options on the input config and it doesn't seem like a single key changed.

Currently, the **mouse** wheel simply does not work at all. If I scroll, no change happens.

Clicking the left button brings up the OSD. The right button exits the video.

Dragging horizontally seeks in the video. Vertically changes brightness.

I.E, I have tried these

-----------
```
MOUSE_BTN3 add volume 2
MOUSE_BTN4 add volume -2
MBTN_MID cycle pause
```
------------
And it didn't change anything. 

I then also tried these,
```
AXIS_UP    seek 10
AXIS_DOWN  seek -10
AXIS_LEFT  seek 5
AXIS_RIGHT seek -5
```
----------------
No change as well.

The input config also doesn't seem to change the behaviour of the remote controller either.  The controller looks like this: [https://images-na.ssl-images-amazon.com/images/I/31FhCIxNZZL.jpg](url) 

The keys also don't seem to change, no matter what I try on the input.conf. 

I figure some keys cannot be changed. But while I see that when I use the remote, using the "input-test", I see the messages of "bound to". The mouse clicks don't seem to register at all.

 By the way, is it possible to add more buttons to the OSD? It currently looks like this for me:
![Screenshot_2019-09-14-18-45-51](https://user-images.githubusercontent.com/30951795/64912371-228e7000-d726-11e9-9b50-63dbcfbbb96f.png)

Any chance to add buttons to jump forward and back, volume control? 

I have searched a lot trying to figure out these issues and finally gave up and decided to ask directly on here.

I use mpv on my PC and it works fine, so would love to figure out what I may be doing wrong here.

Apologies if my message is confusing. I have attached a log life [MPV_log.txt](https://github.com/mpv-android/mpv-android/files/3613024/MPV_log.txt)

Thank you

**Comment by sfan5** [comment]
> Currently, the mouse wheel simply does not work at all. 

That's correct, any mouse interaction is not currently passed to libmpv.

> By the way, is it possible to add more buttons to the OSD?

Planned feature.

**Comment by jbarry786** [comment]
@sfan5 thanks for the reply. 

While mouse interaction is not currently passed, is that something that you see as feasible? 

I am not sure of how much of a niche case I may be, but it feels like it'd be a decent feature to have - it'd be useful to navigate on my android box, android tv.

I have managed to set up 4 keys on the remote control that I referenced.  One interesting thing I notice is that the seep jumps  are not always accurate.
`RIGHT cycle pause`
`LEFT  seek -15`
`UP    seek  15 exact`
`DOWN  seek -60 exact`

When I seek, not limited by keyframes, at times it jumps 1 minute, 2 minutes, even more --- this is strange (which was why I badly needed to edit out the keys so the `exact `would work better).

Interestingly, even when I seek with `UP    seek  15 exact` I still notice that, at times, the jumps can be wild, 1 minute forward or more at times. I tried with different files and the behaviour is different when playing the same files on pc, with the same keybindings.

> Planned feature.

I'm derailing my own original question, which has been satisfied, but let me just give some feedback on a couple of other things.

**1)**

When we play a video and then exit it, we should see the folder or app that launched the file. So for instance, you're using a file explorer and play a video. When you exit it, you should go back to where you were.

Currently,

I have noticed that by checking "Touch gestures", when I run a video and exit it, you will not be lead to the file's path folder. You'll just see mpv's file explorer.

If I uncheck "Touch gestures", then when you exit the video, you will go to the folder where the video is located.

Basically, I think one should be able to have "touch gestures" and also going back to the location where the file was playing from.

**2)**
With regards to the mpv explorer. It could be a good idea to have mpv just displaying folders that contain media files and inside the folders only displaying files that are playable. Currently mpv lists all the folders and all the file types. I know we can set a "default folder" but don't see other options beyond that.

Currently, if you were to use mpv's file explorer, navigating to various folders (for instance external disks) is not very efficient, that is why I think if mpv recognised the folders that actually contain media files, it'd make improve navigation and quick access to the files you want.

Thanks for the feedback sfan,

**Comment by sfan5** [comment]
> While mouse interaction is not currently passed, is that something that you see as feasible?

422c0548d4004ffc214409d6174b37731c93503f enables mouse wheel keybinds
Further mouse buttons are already used by Android for back/forward or "back to homescreen", so those won't be usable by mpv.

> When we play a video and then exit it, we should see the folder or app that launched the file. 

This works fine for me. The fact that it works for you after toggling "touch gestures" is a coincidence.

> With regards to the mpv explorer. It could be a good idea to have mpv just displaying folders that contain media files and inside the folders only displaying files that are playable.

There might be an issue open for this already or at least I've heard this feature suggestion before.
